src/shared/src/impls.rs

```diff
@@ -394,8 +394,12 @@ impl Validate for SplToken {
     fn validate(&self) -> Result<(), candid::Error> {
         use crate::types::MAX_SYMBOL_LENGTH;
         if let Some(symbol) = &self.symbol {
-            if symbol.len() > MAX_SYMBOL_LENGTH {
-                return Err(candid::Error::msg("Symbol too long"));
+            if symbol.chars().count() > MAX_SYMBOL_LENGTH {
```

Comment:
Note: `.len()` counts bytes in a  string, so the "ten snowmen" example in one of our tests has .len() 30.